cmd/solana_exporter/static_test.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestSolanaCollector_Collect_Static(t *testing.T) {
+	collector := createSolanaCollector(
+		&staticRPCClient{},
+		slotPacerSchedule,
+	)
+	prometheus.NewPedanticRegistry().MustRegister(collector)
+
+	testCases := []collectionTest{
+		{
+			Name: "solana_active_validators",
+			ExpectedResponse: `
+# HELP solana_active_validators Total number of active validators by state
+# TYPE solana_active_validators gauge
+solana_active_validators{state="current"} 2
+solana_active_validators{state="delinquent"} 1
+`,
+		},
+		{
+			Name: "solana_validator_activated_stake",
+			ExpectedResponse: `
+# HELP solana_validator_activated_stake Activated stake per validator
+# TYPE solana_validator_activated_stake gauge
+solana_validator_activated_stake{nodekey="aaa",pubkey="AAA"} 49
+solana_validator_activated_stake{nodekey="bbb",pubkey="BBB"} 42
+solana_validator_activated_stake{nodekey="ccc",pubkey="CCC"} 43
+`,
+		},
+		{
+			Name: "solana_validator_last_vote",
+			ExpectedResponse: `
+# HELP solana_validator_last_vote Last voted slot per validator
+# TYPE solana_validator_last_vote gauge
+solana_validator_last_vote{nodekey="aaa",pubkey="AAA"} 92
+solana_validator_last_vote{nodekey="bbb",pubkey="BBB"} 147
+solana_validator_last_vote{nodekey="ccc",pubkey="CCC"} 148
+`,
+		},
+		{
+			Name: "solana_validator_root_slot",
+			ExpectedResponse: `
+# HELP solana_validator_root_slot Root slot per validator
+# TYPE solana_validator_root_slot gauge
+solana_validator_root_slot{nodekey="aaa",pubkey="AAA"} 3
+solana_validator_root_slot{nodekey="bbb",pubkey="BBB"} 18
+solana_validator_root_slot{nodekey="ccc",pubkey="CCC"} 19
+`,
+		},
+		{
+			Name: "solana_validator_delinquent",
+			ExpectedResponse: `
+# HELP solana_validator_delinquent Whether a validator is delinquent
+# TYPE solana_validator_delinquent gauge
+solana_validator_delinquent{nodekey="aaa",pubkey="AAA"} 1
+solana_validator_delinquent{nodekey="bbb",pubkey="BBB"} 0
+solana_validator_delinquent{nodekey="ccc",pubkey="CCC"} 0
+`,
+		},
+		{
+			Name: "solana_node_version",
+			ExpectedResponse: `
+# HELP solana_node_version Node version of solana
+# TYPE solana_node_version gauge
+solana_node_version{version="1.16.7"} 1
+`,
+		},
+	}
+
+	runCollectionTests(t, collector, testCases)
+}
+
+func TestSolanaCollector_WatchSlots_Static(t *testing.T) {
+	// reset metrics before running tests:
+	leaderSlotsTotal.Reset()
+	leaderSlotsByEpoch.Reset()
+
+	collector := createSolanaCollector(
+		&staticRPCClient{},
+		100*time.Millisecond,
+	)
+	prometheus.NewPedanticRegistry().MustRegister(collector)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
```suggestion
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()
```

A bit easier to just defer the cancel in places like this vs some of the places in the code where you have it sprinkled all over some gnarly logic.